### PR TITLE
Add alternative API that takes in file path

### PR DESCRIPTION
**Pull request by zsmoore**
Add api that takes in filePath and initializes visitor to the original file

TODO Need Help:
Tests for ESLint are failing with this change with path resolution
Question on why source and sourceLocatiton are needed, is source not just sourceLocation.readFileSync?

### CODE REVIEW

File: lib/inline-imports.js [hunk 1]
```diff
@@ -86,7 +86,48 @@ function inlineImports(fileContents, options = {}, visited = new Set()) {
...
+ * @property {string} basedir - The base directory to use when resolving import identifiers.  For most platforms this will be the absolute path to the directory containing the graphql query file with imports.
+ * @property {...args} [any] - any other properties on a `ResolverOptions` will be passed through to the resolver.
+ * @param {string} fullFilePath - The graphql document filepath to inline imports into
+ * @param {object} options
+ * @param {(identifier: string, options: ResolverOptions) => string} [options.resolveImport=require.resolve] - a function for resolving imports.  It must support relative paths as well as any package specifiers that may be present as identifiers in the import statements of `fileContents`.  For invalid import identifiers, it should throw an error. Defaults to `resolve.sync` (i.e. treating imported packages as node modules).
+ * @param {ResolverOptions} options.resolveOptions - options that are passed to `options.resolveImport` when resolving imports.
+ * @param {boolean} [throwIfImportNotFound=true] - whether or not to throw for invalid imports, or to silently ignore them.
+ */
+function inlineImportsFromFile(fullFilePath, options = {}) {
```
Comment (hjdivad) [inline lib/inline-imports.js]: My inclination would be to add an option `filePath` to `inlineImports`.

`inlineImportsFromFile` would then just resolve `fullFilePath` and delegate to `inlineImports` rather than duplicate the body of `inlineImports` and also needing to explicitly prepopulate the visited set.


Is there a reason to prefer the current approach?

File: lib/inline-imports.js [hunk 2]
```diff
@@ -86,7 +86,48 @@ function inlineImports(fileContents, options = {}, visited = new Set()) {
...
 }
 
+function getFilename(filePath, options = {}) {
+  const { resolveImport, resolveOptions = {} } = options;
+
+  const { basedir } = resolveOptions;
+  if (typeof basedir !== 'string') {
+    throw new Error('inlineImports requires options.resolverOptions.basedir be set');
+  }
+  const resolve = typeof resolveImport === 'function' ? resolveImport : require('resolve').sync;
```
Comment (hjdivad) [inline lib/inline-imports.js]: This options extraction, validation and defaulting is identical to the code in `linesWithInlinedImportsOf`

Can we extract this to a common utility? `extractOptions` or something

File: lib/inline-imports.js [hunk 3]
```diff
@@ -86,7 +86,48 @@ function inlineImports(fileContents, options = {}, visited = new Set()) {
...
+  return resolve(filePath, resolveOptions);
+}
+
+function getFileContents(filePath, options = {}) {
+  const { fs } = options;
+  const resolvedFs = typeof fs === 'function' ? fs : require('fs');
```
Comment (hjdivad) [inline lib/inline-imports.js:91]: as above i'd like to use a common utility rather than require every use of `fs` to check for a custom `fs` option

File: lib/gather-fragment-imports.js [hunk 4]
```diff
@@ -36,13 +36,24 @@ function gatherFragmentImports({
...
+    });
+  } else {
+    importLinesToInlinedSource = inlineImports.lineToImports(source, {
+      resolveOptions: {
+        basedir: path.dirname(sourceLocation),
+      },
+      resolveImport,
+      throwIfImportNotFound,
+    });
+  }
```
Comment (hjdivad) [inline lib/gather-fragment-imports.js:56]: we should not be doing `path.dirname(undefined`)

`sourceLocation` is a required parameter - let's add a check and throw if it's `undefined` rather than do this branching

File: lib/gather-fragment-imports.js [hunk 5]
```diff
@@ -55,6 +66,7 @@ function gatherFragmentImports({
     lineToFragmentDefinitions.set(lineNumber, fragmentDefinitionsBucket);
   }
 
+  console.log(lineToFragmentDefinitions);
```
Comment (hjdivad) [inline lib/gather-fragment-imports.js]:
```suggestion
```

File: lib/tests/inline-imports-from-file-test.js [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    ]);
+
+    expect(inlineImports.inlineImportsFromFile(basedir + '/custom-resolution.graphql', options)).to
+      .eql(`
+
+fragment CustomResolutionFragment on User {
+  name
+}`);
+  });
+});
```
Comment (hjdivad) [inline lib/tests/inline-imports-from-file-test.js:252]: can we add a test for the edge case of a file A only importing itself?

**Review by hjdivad — CHANGES_REQUESTED** [review]

**Comment by hjdivad** [comment]
@zsmoore this is great!  Thanks for tracking down the initial issue and driving this to completion.  I left some comments inline for a review, lmk if you have any questions.

Your immediate test failures are because your code is invalid in node 12 and earlier, but we should just bump our CI #16

**Comment by hjdivad** [comment]
@zsmoore got you passed the linting issues now you have actual test failures to dig  into
